### PR TITLE
Add main viewModelTest

### PR DESCRIPTION
**Pull request by bibiosm98**
Add main mainViewModel tests

### CODE REVIEW

File: app/src/test/java/com/example/internapp/MainViewModelTest.kt [hunk 1]
```diff
@@ -3,72 +3,103 @@ package com.example.internapp
...
+                "Description IronMan2",
+                Thumbnail("path to res", "JPG"),
+                Creators(0, emptyList()),
+                emptyList()
+            )
+        )
         FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext())
 
         val marvelApiRepository = mockk<MarvelApiRepository>()
+        every {
```
Comment (kamiltomczykdv) [inline app/src/test/java/com/example/internapp/MainViewModelTest.kt]: You could use coEvery instead

File: app/src/test/java/com/example/internapp/MainViewModelTest.kt [hunk 2]
```diff
@@ -3,72 +3,103 @@ package com.example.internapp
...
+            runBlocking {
+                delay(3000)
```
Comment (kamiltomczykdv) [inline app/src/test/java/com/example/internapp/MainViewModelTest.kt]: What is this delay for?

Comment (bibiosm98) [reply app/src/test/java/com/example/internapp/MainViewModelTest.kt]: Just "simulate" real delay while request API.

File: app/src/test/java/com/example/internapp/MainViewModelTest.kt [hunk 3]
```diff
@@ -96,47 +127,62 @@ class MainViewModelTest{
...
+        assertEquals(mainViewModel.comicList.value?.get(1), mainViewModel.selectedComic.value)
+        assertThat(mainViewModel.selectedComic.value, `is`(not(nullValue())))
+    }
+
+    @Test
+    fun signOutUser_userLoggedOut_returnsOnWaiting() {
+        mainViewModel.signOutUser()
+        assertEquals(mainViewModel.uiState.value, UIState.OnWaiting)
+        val uiState = mainViewModel.uiState.getOrAwaitValue()
+        assertThat(uiState, `is`(not(nullValue())))
```
Comment (kamiltomczykdv) [inline app/src/test/java/com/example/internapp/MainViewModelTest.kt]: I don't understand it, isn't it possible to check whether uiState equals particular value?

File: app/src/test/java/com/example/internapp/MainViewModelTest.kt (hunk 2 above)
Comment (kamiltomczykdv) [reply app/src/test/java/com/example/internapp/MainViewModelTest.kt]: It's no necessary

**Review by kamiltomczykdv — APPROVED** [review]

**Pull request merged**